ShortenLink/Scenes/ShortenLink/ShortenLinkViewModel.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  ShortenLinkViewModel.swift
+//  ShortenLink
+//
+//  Created by Bliss on 22/11/21.
+//
+
+import Foundation
+import RxCocoa
+import RxSwift
+
+protocol ShortenLinkViewModelType {
+
+    var input: ShortenLinkViewModelInput { get }
+    var output: ShortenLinkViewModelOutput { get }
+}
+
+protocol ShortenLinkViewModelOutput {
+
+    var linkShortenSuccess: Driver<Bool> { get }
+}
+
+protocol ShortenLinkViewModelInput {
+
+    func shortenLink(link: String)
+}
+
+final class ShortenLinkViewModel: ShortenLinkViewModelType {
+
+    var linkShortenSuccess: Driver<Bool>
+
+    var input: ShortenLinkViewModelInput { self }
+    var output: ShortenLinkViewModelOutput { self }
+
+    private var shortenLinkTrigger = PublishRelay<String>()
+    private var createShortenLinkUseCase: CreateShortenLinkUseCaseProtocol
+
+    init(createShortenLinkUseCase: CreateShortenLinkUseCaseProtocol) {
+        self.createShortenLinkUseCase = createShortenLinkUseCase
+
+        linkShortenSuccess = shortenLinkTrigger
+            .flatMapLatest { input -> Driver<Bool> in
+                createShortenLinkUseCase
+                    .createShortenLink(input)
+                    .asObservable().materialize()
+                    .filter { $0.element != nil }
+                    .map {
+                        let pasteboard = NSPasteboard.general
+                        // TODO: Use Constant for link prefix
+                        pasteboard.setGeneralString("https://l.nimblehq.co/\($0.element?.alias ?? "")")
+                        return true
+                    }
+                    .asDriver(onErrorJustReturn: false)
+            }
+            .asDriver(onErrorJustReturn: false)
+    }
+}
+
+extension ShortenLinkViewModel: ShortenLinkViewModelInput {
+
+    func shortenLink(link: String) {
```

Comment:
You won't need this and the `shortenLinkTrigger` if the input of ViewModel is PublishRelay